components/utils/src/links.rs

```diff
@@ -0,0 +1,42 @@
+use regex::Regex;
+
+
+pub fn anchor_id_checks(anchor:&str) -> Regex {
+    Regex::new(
+        &format!(r#" (?i)(id|ID|name|NAME) *= *("|')*{}("|'| |>)+"#, anchor)
```

Comment:
Since we're setting the `?i` flag, we can get rid of the multiple cases in the first part of the regex